File: src/main/java/factory/ScheduleViewFactory.java
package factory;

import app.FXMLPaths;
import control.schedule.WeekScheduleController;
import javafx.scene.Node;

import java.io.IOException;

public class ScheduleViewFactory<C extends WeekScheduleController> extends ViewFactory<C> {

    public ScheduleViewFactory(C controller) {
        super(FXMLPaths.GenericSchedule());
        this.controller = controller;
    }

    @Override
    public Node load() throws IOException {
        return super.load(controller);
    }
}

File: src/main/scala/factory/ScheduleViewFactory.scala
package factory

import java.io.IOException

import app.FXMLPaths
import control.schedule.WeekScheduleController
import javafx.scene.Node

class ScheduleViewFactory[C <: WeekScheduleController](controller: C) extends ViewFactory[C](FXMLPaths.GenericSchedule) {

    @throws[IOException]
    override def load: Node = super.load(controller)
}